modules/ducktests/tests/ignitetest/tests/discovery_test.py

```diff
@@ -71,40 +72,63 @@ class DiscoveryTest(IgniteTest):
     """
     NUM_NODES = 7
 
-    FAILURE_DETECTION_TIMEOUT = 2000
+    FAILURE_DETECTION_TIMEOUT = 1000
 
     DATA_AMOUNT = 5_000_000
 
     WARMUP_DATA_AMOUNT = 10_000
 
+    NETFILTER_SAVED_SETTINGS = os.path.join(IgniteTest.TEMP_PATH_ROOT, "discovery_test", "netfilter.bak")
+
     @cluster(num_nodes=NUM_NODES)
     @ignite_versions(str(DEV_BRANCH), str(LATEST_2_8))
-    @matrix(kill_coordinator=[False, True],
-            nodes_to_kill=[1, 2],
+    @matrix(nodes_to_kill=[1, 2],
             load_type=[ClusterLoad.NONE, ClusterLoad.ATOMIC, ClusterLoad.TRANSACTIONAL])
-    def test_node_fail_tcp(self, ignite_version, kill_coordinator, nodes_to_kill, load_type):
+    def test_nodes_fail_notseq_tcp(self, ignite_version, nodes_to_kill, load_type):
         """
-        Test nodes failure scenario with TcpDiscoverySpi.
-        :param load_type: How to load cluster during the test: 0 - no loading; 1 - do some loading; 2 - transactional.
+        Test nodes failure scenario with TcpDiscoverySpi not allowing nodes to fail in a row.
         """
-        test_config = DiscoveryTestConfig(version=IgniteVersion(ignite_version), kill_coordinator=kill_coordinator,
-                                          nodes_to_kill=nodes_to_kill, load_type=load_type, with_zk=False)
+        test_config = DiscoveryTestConfig(version=IgniteVersion(ignite_version), nodes_to_kill=nodes_to_kill,
+                                          load_type=load_type, sequential_failure=False)
+
+        return self._perform_node_fail_scenario(test_config)
+
+    @cluster(num_nodes=NUM_NODES)
+    @ignite_versions(str(DEV_BRANCH), str(LATEST_2_8))
+    @matrix(load_type=[ClusterLoad.NONE, ClusterLoad.ATOMIC, ClusterLoad.TRANSACTIONAL])
+    def test_2_nodes_fail_seq_tcp(self, ignite_version, load_type):
+        """
+        Test 2 nodes sequential failure scenario with TcpDiscoverySpi.
+        """
+        test_config = DiscoveryTestConfig(version=IgniteVersion(ignite_version), nodes_to_kill=2, load_type=load_type,
+                                          sequential_failure=True)
 
         return self._perform_node_fail_scenario(test_config)
 
     @cluster(num_nodes=NUM_NODES + 3)
     @version_if(lambda version: version != V_2_8_0)  # ignite-zookeeper package is broken in 2.8.0
     @ignite_versions(str(DEV_BRANCH), str(LATEST_2_8))
-    @matrix(kill_coordinator=[False, True],
-            nodes_to_kill=[1, 2],
+    @matrix(nodes_to_kill=[1, 2],
             load_type=[ClusterLoad.NONE, ClusterLoad.ATOMIC, ClusterLoad.TRANSACTIONAL])
-    def test_node_fail_zk(self, ignite_version, kill_coordinator, nodes_to_kill, load_type):
+    def test_nodes_fail_notseq_zk(self, ignite_version, nodes_to_kill, load_type):
```

Comment:
If linter allowed longer method name, may be call it not_sequential_zk?

Reply:
Fixed